src/harness/BenchmarkDotNet.Extensions/CommandLineOptions.cs

```diff
@@ -53,6 +54,44 @@ public static List<string> ParseAndRemoveStringsParameter(List<string> argsList,
             return argsList;
         }
 
+        public static List<string> ParseAndRemovePairsParameter(List<string> argsList, string parameter, out Dictionary<string, object> pairValues)
+        {
+            int parameterIndex = argsList.IndexOf(parameter);
+            pairValues = new Dictionary<string, object>();
+
+            if (parameterIndex + 1 < argsList.Count)
+            {
+                while (parameterIndex + 1 < argsList.Count && !argsList[parameterIndex + 1].StartsWith("-"))
+                {
+                    string[] pair = argsList[parameterIndex + 1].Split(':');
```

Comment:
No, if that is a valid concern, we will need to identify which character/character combo works as a valid separator between a parameter name and a value.  Spaces are already being used to separate parameter blocks.

Reply:
Do we know which characters would not be legal in the name? Maybe one of those could be used as the separator.

Reply:
The problem is not the name of the parameter, but the value. For strings it can be almost anything. For now, we can leave it as is, and if we come across a value that has this we can fix it then.

Reply:
`Split()` takes a second argument that will limit the number of groups to split into.

`.Split(':',2)` will preserve ALL of whatever's on the right side of the colon.

Illustration: https://dotnetfiddle.net/EAuVmo